helpers/mime.js

```diff
@@ -2,28 +2,54 @@
 
 const mime = require('mime-types');
 
-const FORCE_EXTENSION_MAPPING = {
-    yidf: 'txt',
-    state: 'txt',
-    diff: 'txt',
-    xml: 'txt' // FIXME: Chrome is not displaying xml files
-};
+const KNOWN_FILE_NAMES_IN_TEXT_FORMAT = ['dockerfile', 'makefile'];
+const KNOWN_FILE_EXTS_IN_TEXT_FORMAT = [
+    'js',
+    'c',
+    'cpp',
+    'cs',
+    'dtd',
+    'h',
+    'm',
+    'java',
+    'lua',
+    'pl',
+    'go',
+    'py',
+    'tox',
+    'env',
+    'sh',
+    'vb',
+    'swift',
+    'yidf',
+    'state',
+    'diff',
+    'deps',
+    'xml'
+];
 
 /**
- * getMimeFromFileExtension
+ * Get MIME types from file name and file extension
+ * @method getMimeFromFileName
  * @param  {String} fileExtension  File extension (e.g. css, txt, html)
- * @return {String} text/html
+ * @param  {String} fileName       File name      (e.g. dockerfile, main)
+ * @return {String} MIME Type      eg. text/html, text/plain
  */
-function getMimeFromFileExtension(fileExtension) {
-    return mime.lookup(FORCE_EXTENSION_MAPPING[fileExtension] || fileExtension) || '';
+function getMimeFromFileName(fileExtension, fileName = '') {
+    if (KNOWN_FILE_NAMES_IN_TEXT_FORMAT.includes(fileName.toLowerCase())) {
```

Comment:
instead of a static list, can be generalize this to have any fileName which ends with `file` ?

Reply:
Like dockerfile, makefile? I can update the logic for that.